bgs/fedmgr.go

```diff
@@ -389,15 +440,22 @@ func (s *Slurper) handleConnection(ctx context.Context, host *models.PDS, con *w
 		},
 	}
 
+	s.LimitMux.RLock()
```

Comment:
a general code cleanliness thing im a fan of is if you have a lock in the middle of your function, it probably wants to be in its own function